clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -30,19 +30,24 @@
 
 public class MetadataRequest extends AbstractRequest {
 
+    private static final String TOPICS_KEY_NAME = "topics";
+    private static final String ALLOW_AUTO_TOPIC_CREATION_KEY_NAME = "allow_auto_topic_creation";
+
     public static class Builder extends AbstractRequest.Builder<MetadataRequest> {
         private static final List<String> ALL_TOPICS = null;
 
         // The list of topics, or null if we want to request metadata about all topics.
         private final List<String> topics;
+        private final boolean allowAutoTopicCreation;
 
         public static Builder allTopics() {
-            return new Builder(ALL_TOPICS);
+            return new Builder(ALL_TOPICS, false);
```

Comment:
The behaviour of `ALL_TOPICS` is that auto creation never happens. So, I had set it to `false` to make that obvious. All the code handles this properly. The one downside is that if you serialize a version that doesn't have this field and then deserialize, the deserialized version will have the boolean as `true`. This may be confusing, so I will change it to `true` like you suggested.